### ISSUE TITLE
Avoid to parse an ontology source file that is the same that previous version, even if retrieved by cron job

### ISSUE THREAD

**Issue by jonquet** [agroportal/project-management#171]
This issue is a long term fix for problems such as the ones in #167 

We need to implement a mechanism that explicitly double check that the source file downloaded is different from the source file of the previous version (if exist). 
This mechanism must be added to the automatic pull script that check file changes with HTTP headers.

**Comment by syphax-bouazzouni** [comment]
after checking the code to know more about how is the ncbo_cron job figuring out, that a new version of an ontology was released, we  found that **it's not looking for the http header** but download every day the ontologies and hash it to  compare it with the local ones (see code below source : https://github.com/ontoportal-lirmm/ncbo_cron/blob/master/lib/ncbo_cron/ontology_pull.rb#L54)

![image](https://user-images.githubusercontent.com/29259906/148270744-855d670b-d6da-4c75-ba7e-2c4dbd07257d.png)

In summary here is the process of pulls checking. Every day at 18:00,  for each ontology we :
1. [Check if the URI is accessible and respond](https://github.com/ontoportal-lirmm/ontologies_linked_data/blob/master/lib/ontologies_linked_data/models/ontology_submission.rb#L2330) 
2. [Download the remote file, Hash it  and compare it with the local file hash. ](https://github.com/ontoportal-lirmm/ncbo_cron/blob/master/lib/ncbo_cron/ontology_pull.rb#L54)

**Comment by syphax-bouazzouni** [comment]
So @jonquet, the requirement of this issue need to be updated or to be closed.

**Comment by jonquet** [comment]
Now that we see how the script works we can close this issue as it is already implemented. 
In the future, we might create another issue to enhance the script so that it does not pull the fils if HTTP headers do no indicate a change since the last pulled one.
